habitat-lab/habitat/config/default_structured_configs.py

```diff
@@ -1442,6 +1442,12 @@ class SimulatorConfig(HabitatBaseConfig):
     # If the number of agents is greater than one,
     # then agents_order has to be set explicitly.
     agents_order: List[str] = MISSING
+
+    # Simulator should use default navmesh settings from agent config
+    default_agent_navmesh: bool = True
```

Comment:
Hi @aclegg3, thank you for the excellent series of refactoring!

I completely understand the importance of this motivation. However, I would like to suggest setting `default_agent_navmesh` to `False` until the new feature has undergone thorough testing. Currently, when running `python examples/example.py`, there are some additional error messages that appear, such as `AttributeError: 'habitat_sim._ext .habitat_sim_bindings.SimulatorConfiguration' object has no attribute 'navmesh_settings'`.

Reply:
@wangyenjen thanks for the feedback. 

From the error it looks like you may not have updated locally to habitat-sim `main`. That attribute was added in habitat-sim [2111](https://github.com/facebookresearch/habitat-sim/pull/2111) and habitat-lab [1351](https://github.com/facebookresearch/habitat-lab/pull/1351). Try again with nightly conda build or source build on main. :) 

Note that this behavior (`default_agent_navmesh=True`) mimics the expected behavior from before the above PRs.

Reply:
@aclegg3, I understand. In fact, I also tried the nightly conda build for the main branch but encountered the same error message. Perhaps I should attempt to build the current main branch for testing purposes. Thank you very much!